host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -1471,7 +1471,7 @@ static void encoder_buffer_callback(MMAL_PORT_T *port, MMAL_BUFFER_HEADER_T *buf
    {
       MMAL_STATUS_T status;
 
-      new_buffer = mmal_queue_get(pData->pstate->encoder_pool->queue);
+      new_buffer = pData? mmal_queue_get(pData->pstate->encoder_pool->queue) : NULL;
```

Comment:
I do see that there is a check on pData above, but what is the actual sequence of events that results in pData being NULL? To me it sounds like that is the thing that needs to be fixed as so much depends on being able to link back to the state.
userdata is set before the mmal_port_enable call, so there should never be any calls to this before it is set, it is never cleared by the app, and the core should be leaving it alone as it is app data.

If this one does need fixing, then so does the identical call in splitter_buffer_callback.